From 00ba31795d462a1f0eeb2bcd2c5ccdf569cfb067 Mon Sep 17 00:00:00 2001
From: mpvader <mvader@victronenergy.com>
Date: Sun, 6 Jul 2025 10:11:28 +0200
Subject: [PATCH 4/4] package.json: remove typedoc-signalk-theme devDependency

This package is not needed in the Venus OS install (its a devDepencency),
but for some reason NPM wants to do something with it anyway and fails.

256 verbose stack HttpErrorGeneral: 404 Not Found - GET https://registry.npmjs.org/typedoc-signalk-theme - Not found
5256 verbose stack     at /media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/npm-registry-fetch/lib/check-response.js:95:15
5256 verbose stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
5256 verbose stack     at async RegistryFetcher.packument (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/pacote/lib/registry.js:90:19)
5256 verbose stack     at async RegistryFetcher.manifest (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/pacote/lib/registry.js:128:23)
5256 verbose stack     at async #fetchManifest (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1199:20)
5256 verbose stack     at async #nodeFromEdge (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1037:19)
5256 verbose stack     at async #buildDepStep (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:901:11)
5256 verbose stack     at async Arborist.buildIdealTree (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:181:7)
5256 verbose stack     at async Promise.all (index 1)
5256 verbose stack     at async Arborist.reify (/media/venusbuilds/venus/build/tmp-glibc/work/einstein-ve-linux-gnueabi/signalk-server/2.14.4/recipe-sysroot-native/usr/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:131:5)
5257 verbose statusCode 404
5258 verbose pkgid typedoc-signalk-theme@^0.3.0
5259 error code E404
5260 error 404 Not Found - GET https://registry.npmjs.org/typedoc-signalk-theme - Not found
5261 error 404
5262 error 404  'typedoc-signalk-theme@^0.3.0' is not in this registry.
5263 error 404
5264 error 404 Note that you can also install from a
5265 error 404 tarball, folder, http url, or git url.
---
 package.json | 1 -
 1 file changed, 1 deletion(-)

diff --git a/package.json b/package.json
index b52e00bd..4945ae96 100644
--- a/package.json
+++ b/package.json
@@ -166,7 +166,6 @@
     "ts-node": "^10.9.1",
     "typedoc": "^0.28.1",
     "typedoc-plugin-mdn-links": "^5.0.1",
-    "typedoc-signalk-theme": "^0.3.0",
     "typescript": "^5.8.2",
     "typescript-eslint": "^8.29.1"
   },
-- 
2.34.1

